hadoop-ozone/ozone-recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTask.java

```diff
@@ -0,0 +1,198 @@
+    }
+  }
+
+  /**
+   * Read the Keys from OM snapshot DB and calculate the upper bound of
+   * File Size it belongs to.
+   *
+   * @param omMetadataManager OM Metadata instance.
+   * @return Pair
+   */
+  @Override
+  public Pair<String, Boolean> reprocess(OMMetadataManager omMetadataManager) {
+    LOG.info("Starting a 'reprocess' run of FileSizeCountTask.");
+
+    Table<String, OmKeyInfo> omKeyInfoTable = omMetadataManager.getKeyTable();
+    try (TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>>
+        keyIter = omKeyInfoTable.iterator()) {
+      while(keyIter.hasNext()) {
+        Table.KeyValue<String, OmKeyInfo> kv = keyIter.next();
+        countFileSize(kv.getValue());
+      }
+      populateFileCountBySizeDB();
+
+    } catch (IOException ioEx) {
+    LOG.error("Unable to populate Container Key Prefix data in Recon DB. ", ioEx);
+    return new ImmutablePair<>(getTaskName(), false);
+    }
+
+    LOG.info("Completed a 'reprocess' run of FileSizeCountTask.");
+    return new ImmutablePair<>(getTaskName(), true);
+  }
+
+  @Override
+  protected Collection<String> getTaskTables() {
+    return tables;
+  }
+
+  /**
+   * Read the Keys from update events and update the count of files
+   * pertaining to a certain upper bound.
+   *
+   * @param events Update events - PUT/DELETE.
+   * @return Pair
+   */
+  @Override
+  Pair<String, Boolean> process(OMUpdateEventBatch events) {
+    LOG.info("Starting a 'process' run of FileSizeCountTask.");
+    Iterator<OMDBUpdateEvent> eventIterator = events.getIterator();
+    while (eventIterator.hasNext()) {
+      OMDBUpdateEvent<String, OmKeyInfo> omdbUpdateEvent = eventIterator.next();
+      String updatedKey = omdbUpdateEvent.getKey();
+      OmKeyInfo updatedValue = omdbUpdateEvent.getValue();
+
+      try{
+        switch (omdbUpdateEvent.getAction()) {
+          case PUT:
+            updateCountForKey(updatedValue, "PUT");
+            break;
+
+          case DELETE:
+            updateCountForKey(updatedValue, "DELETE");
+            break;
+
+          default: LOG.debug("Skipping DB update event : " + omdbUpdateEvent
+                  .getAction());
+        }
+
+      } catch (IOException e) {
+        LOG.error("Unexpected exception while updating key data : {} ",
+                updatedKey, e);
+        return new ImmutablePair<>(getTaskName(), false);
+      }
+    }
+    LOG.info("Completed a 'process' run of FileSizeCountTask.");
+    return new ImmutablePair<>(getTaskName(), true);
+  }
+
+  /**
+   * Calculate the bin index based on size of the Key.
+   *
+   * @param dataSize Size of the key.
+   * @return int bin index in upperBoundCount
+   */
+  private int findBinIndex(long dataSize) {
+    long bin = dataSize/ minFileSize;
+    if(bin == 0) {
+      return 0;
+    }
+    return (int) Math.log10(bin) + 1;
+  }
+
+  private void countFileSize(OmKeyInfo omKeyInfo) {
+    upperBoundCount[findBinIndex(omKeyInfo.getDataSize())]++;
```

Comment:
It is highly unlikely, But just to make sure we don't hit ArrayOutOfBoundsException for a file, maybe we can have the last bin (# 17) as an open bin for all files > 1PB.